### PR TITLE
Enable screens list to be modified using Mavo

### PR DESCRIPTION
**Pull request by efeichen**
Instead of sending pull requests or file Github issues, visitors can now use Mavo to log into their Github account, edit the list right on the website, and save to send edit suggestions (pull requests).

Potential improvements for the future:
1. Allow the list to be sorted (asc and desc) by the table headers.
2. Show "no result" when no search result is found.
3. Search re-updates result when deleting characters on search bar (right now you have to clear it for you to search again and get results).

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -66,13 +67,11 @@ <h1>
...
-
-	<section id="devices">
+	<section id="devices" mv-app mv-storage="https://github.com/LeaVerou/dpi/blob/gh-pages/screens.json" class="mv-autoedit">
 		<h1>Known screens</h1>
-
-		<input type="search" placeholder="Search…" id="search" />
-
+		<input type="search" placeholder="Search…" id="search" property mv-storage="none"/>
 		<div id="screens">
+			<div class="mv-bar mv-ui" mv-bar="no-login"></div>
```
Comment (LeaVerou) [inline index.html:74]: You don't need an element for the Mavo bar if you don't plan to customize it. You can place the `mv-bar` attribute on the Mavo root.

File: assets/js/dpi.js [hunk 2]
```diff
@@ -70,6 +71,7 @@ $u.xhr({
 		tbody.appendChild(fragment);
 	}
 });
+*/
```
Comment (LeaVerou) [inline assets/js/dpi.js]: No need to comment out unneeded JS, just delete it. That's why we have version control: If we need this code at some point, we can go back in history and get it. :)
Also, I'm pretty sure there's more code that could be removed. E.g. the code that did the searching or the `deviceRow()` function.

File: index.html [hunk 3]
```diff
@@ -84,13 +83,17 @@ <h1>Known screens</h1>
 					</tr>
 				</thead>
 				<tbody>
-					<tr>
-						<td colspan="4">Loading…</td>
+					<tr property="screen" mv-multiple mv-if="!search or search(lowercase(name), lowercase(search))>=0 or search(d, search)>=0 or search(w, search)>=0 or search(h, search)>=0 or search(dpi, search)>=0 or search(dppx, search)>=0">
```
Comment (LeaVerou) [inline index.html]: I think this long expression would be much more readable if it were formatted as multiple lines.

**Review by LeaVerou — COMMENTED** [review]
Overall looks great! See comments for a few minor tweaks.

File: index.html (hunk 1 above)
Comment (efeichen) [reply index.html:74]: I thought it looked weird that the Mavo bar appeared on top of the "known screens" heading.

**Comment by efeichen** [comment]
Just made the changes you requested. As for Mavo integration on the calculator above, I could make a separate pull request. 👍 

P.S. Might have accidentally clicked "Close & Comment" yesterday that's why the pull request got closed.

**Comment by LeaVerou** [comment]
Thank you!
Just noticed that the original website had a "No Results" message when there were no results. Can we keep that? IIRC that was one of your own todos as well :)
You could do this by assigning the `mv-if` condition to a hidden property (`<meta property="foobar" content="[expression]" />`) and then having the message as a separate `<tr>` with an `mv-if` that counts that property.

**Comment by efeichen** [comment]
Is the `tr` counting the number of `meta` elements present? Or the number of "results" shown on the page (kind of like what your original JS function did).
A little confused on how the hidden property will help me with this.

**Comment by LeaVerou** [comment]
Looks good, merged!

**Pull request merged**